config.nix

```diff
@@ -303,13 +303,17 @@ let
       in
         lib.unique (lib.concatMap expandMember listedMembers);
 
-    patchedSources =
+    copySourcesFrom = if copySources != [] then src else null;
```

Comment:
Why not always pass `src` here?

Reply:
I originally always passed `src`, but it turns out it would add a dependency on `src` all the time, which breaks the two-step compilation process and forces all dependencies to always be rebuilt.

Actually, this is probably also an issue in the implementation for path-based patch dependencies things, that also use `patchedSources`: as soon as there's a patch the deps-dev derivation depends on the src derivation, which breaks deps-dev caching. Not having had any idea to solve this issue, I just made sure there's no regression when not passing in `copySources`.

Should I eg. add a comment about that here?

Reply:
Yes that would be great! Still wrapping my head around the caching issue.